src/test/java/io/jenkins/plugins/coverage/model/CoverageEvaluatorTest.java

```diff
@@ -0,0 +1,103 @@
+        QualityGate lineQualityGate = new QualityGate(0.95, 0.80, CoverageMetric.LINE);
+        qualityGates.add(lineQualityGate);
+        qualityGates.add(new QualityGate(0.95, 0.80, CoverageMetric.METHOD));
+        qualityGates.add(new QualityGate(0.95, 0.80, CoverageMetric.CLASS));
+        evaluator.addAll(qualityGates);
+        QualityGateStatus qualityGateStatus = evaluator.evaluate(coverageNode, logger);
+
+        assertThat(qualityGates).hasSize(3);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.SUCCESSFUL);
+        assertThat(logger.getMessages()).containsExactly("-> SUCCESSFUL - QualityGate");
+
+        qualityGates.remove(lineQualityGate);
+        logger.clear();
+        qualityGateStatus = evaluator.evaluate(coverageNode, logger);
+
+        assertThat(qualityGates).hasSize(2);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.SUCCESSFUL);
+        assertThat(logger.getMessages()).containsExactly("-> SUCCESSFUL - QualityGate");
+
+    }
+
+    @Test
+    void shouldFail() {
+        Logger logger = new Logger();
+        CoverageNode coverageNode = CoverageNodeConverter.convert(readResult("jacoco-analysis-model.xml"));
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        evaluator.add(new QualityGate(0.99, 0.80, CoverageMetric.FILE));
+        evaluator.add(new QualityGate(0.99, 0.80, CoverageMetric.METHOD));
+        evaluator.add(new QualityGate(0.99, 0.91, CoverageMetric.BRANCH));
+        QualityGateStatus qualityGateStatus = evaluator.evaluate(coverageNode, logger);
+
+        assertThat(logger.getMessages()).containsExactly(
+                "-> FAILED - QualityGate: " + CoverageMetric.BRANCH.getName() + " - warn/fail/actual: " + 0.99 + "/"
+                        + 0.91 + "/"
+                        + 0.89);
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.FAILED);
+    }
+
+    @Test
+    void shouldBeInactive() {
+        Logger logger = new Logger();
+        CoverageNode coverageNode = CoverageNodeConverter.convert(readResult("jacoco-analysis-model.xml"));
+        CoverageEvaluator evaluator = new CoverageEvaluator();
+        QualityGateStatus qualityGateStatus = evaluator.evaluate(coverageNode, logger);
+
+        assertThat(qualityGateStatus).isEqualTo(QualityGateStatus.INACTIVE);
+        assertThat(logger.getMessages()).containsExactly("-> INACTIVE - No quality gate defined");
+    }
+
+    /**
+     * Logger for the tests that provides a way verify and clear the messages.
+     */
+    private static class Logger implements FormattedLogger {
+        private final List<String> messages = new ArrayList<>();
+
+        @Override
+        public void print(final String format, final Object... args) {
+            messages.add(String.format(format, args));
+        }
+
+        List<String> getMessages() {
+            return messages;
+        }
+
+        void clear() {
+            messages.clear();
+        }
+    }
+}
```

Comment:
In total, the tests are well written and cover most important paths. (One is missing: a quality gate that has no match in the node).